packages/vite/src/node/ssr/index.ts

```diff
@@ -50,11 +50,9 @@ export function resolveSSROptions(
   let format: SSRFormat = 'esm'
   let target: SSRTarget = 'node'
   if (buildSsrCjsExternalHeuristics) {
-    if (ssr) {
-      format = 'cjs'
-    } else {
+    format = 'cjs'
+    if (!ssr) {
```

Comment:
`target` is already `'node'`, we can remove this condition too. Note: We plan to remove CJS external heuristics support in Vite 5. If you are using it for some reason, please comment on https://github.com/vitejs/vite/discussions/12466. We hope to move to ESM only for SSR as all services now properly support ESM.